lib/verifier.py

```diff
@@ -167,10 +167,35 @@ def _raise_if_valid_tx(cls, raw_tx: str):
         tx = Transaction(raw_tx)
         try:
             tx.deserialize()
-        except:
-            pass
-        else:
-            raise InnerNodeOfSpvProofIsValidTx()
+        except SerializationError:
+            return
+
+        # Besides deserializing, we perfom additional checks to reduce the
+        # chance that a legitimate merkle node give false positive.
+
+        # A legitimate txn of 64 bytes can only have 1 input and 1 output.
+        try:
+            (txin,) = tx.inputs()
+            (txout,) = tx.outputs()
+        except ValueError:
+            return
+
+        # Txes can be at most 1 MB and and each output requires >= 9 bytes;
+        # Thus prevout_n must <= 111111 for a real spend; it may also be
+        # max uint32 for a coinbase.
+        if 111111 < txin['prevout_n'] < 0xff_ff_ff_ff:
```

Comment:
Note: these _ digit separators do require python 3.6. We good on that?

Reply:
Yes.. it's a requirement now .. the app aborts if Python 3.5.  Also it's on the front page of the github and all our builds are 3.6 so it's pretty strongly enforced.

Also the app crashes if you're on 3.5 anyway. :)  We've been using 3.6 code for a few months now.